src/Symfony/Bundle/TwigBundle/ErrorRenderer/TwigHtmlErrorRenderer.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\ErrorRenderer;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\ErrorRenderer\ErrorRenderer\HtmlErrorRenderer;
+use Symfony\Component\ErrorRenderer\Exception\FlattenException;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Twig\Environment;
+use Twig\Error\LoaderError;
+use Twig\Loader\ExistsLoaderInterface;
+
+/**
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class TwigHtmlErrorRenderer extends HtmlErrorRenderer
+{
+    private $twig;
+    private $debug;
+
+    public function __construct(Environment $twig, RequestStack $requestStack, bool $debug = false, string $charset = null, $fileLinkFormat = null, string $projectDir = null, LoggerInterface $logger = null)
+    {
+        $this->twig = $twig;
+        $this->debug = $debug;
+
+        parent::__construct($debug, $charset, $fileLinkFormat, $projectDir, $requestStack, $logger);
+    }
+
+    public function render(FlattenException $exception): string
+    {
+        if ($exception->getHeaders()['X-Debug'] ?? $this->debug) {
+            return parent::render($exception);
+        }
+
+        $template = $this->findTemplate($exception->getStatusCode());
+
+        if (null === $template) {
+            return parent::render($exception);
+        }
+
+        return $this->twig->render($template, [
+            'legacy' => false, // to be removed in 5.0
+            'exception' => $exception,
+            'status_code' => $exception->getStatusCode(),
+            'status_text' => $exception->getTitle(),
+        ]);
+    }
+
+    private function findTemplate(int $statusCode): ?string
+    {
+        $template = sprintf('@Twig/Exception/error%s.html.twig', $statusCode);
+        if ($this->templateExists($template)) {
+            return $template;
+        }
+
+        $template = '@Twig/Exception/error.html.twig';
+        if ($this->templateExists($template)) {
+            return $template;
+        }
+
+        return null;
+    }
+
+    /**
+     * @deprecated since Symfony 4.4
```

Comment:
it's a privat method that does not need to be deprecated. maybe you wanted to say "to be removed in 5.0"? but what would it be replaced with?